openlibrary/plugins/upstream/covers.py

```diff
@@ -36,6 +36,10 @@ def POST(self, key):
         if not book:
             raise web.notfound("")
 
+        user = accounts.get_current_user()
+        if user.is_read_only():
```

Comment:
We may want to check `user` here as well;
```suggestion
        if user and user.is_read_only():
```